File: src/nim/slate/elements.nim
#:___________________________________________________________________
#  *Slate  |  Copyright (C) Ivan Mar (sOkam!)  |  LGPLv3 or higher  :
#:___________________________________________________________________
## @fileoverview Cable connector to all Element modules
#_______________________________________________________|
import ./element/general   ; export general
import ./element/statement ; export statement
import ./element/procs     ; export procs
import ./element/vars      ; export vars
import ./element/pragmas   ; export pragmas
import ./element/calls     ; export calls
import ./element/affixes   ; export affixes
import ./element/types     ; export types
import ./element/arrays    ; export arrays


#_______________________________________
# @note
#  To have access to the Slate syntax rules extensions:
#  1. Compile with -d:useExtras
#  2. or import this module explicitly in the files where you need it
when defined(useExtras):
  import ./element/extras ; export extras
#_______________________________________

File: src/slate/elements.zig
//:___________________________________________________________________
//  *Slate  |  Copyright (C) Ivan Mar (sOkam!)  |  LGPLv3 or higher  :
//:___________________________________________________________________
//! @fileoverview Cable connector to all of the element modules
//______________________________________________________________|
pub const Ident    = @import("./element/ident.zig").Ident;
pub const Type     = @import("./element/type.zig").Type;
pub const Data     = @import("./element/data.zig").Data;
pub const Pragma   = @import("./element/pragma.zig").Pragma;
pub const Expr     = @import("./element/expression.zig").Expr;
pub const Stmt     = @import("./element/statement.zig").Stmt;
//__________________________________________________________
pub const Node     = @import("./element/node.zig").Node;
pub const Root     = @import("./element/root.zig").Root;
pub const Proc     = @import("./element/proc.zig").Proc;
pub const Variable = @import("./element/statement.zig").Stmt.Variable;

